tests/hlsl-intrinsic/matrix-select.slang

```diff
@@ -0,0 +1,94 @@
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHK): -dx12 -shaderobj -output-using-type -xslang -matrix-layout-column-major
+//TEST(compute):COMPARE_COMPUTE(filecheck-buffer=CHK): -dx12 -shaderobj -output-using-type -xslang -matrix-layout-row-major
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+RWStructuredBuffer<float> outputBuffer;
+
+[numthreads(1, 1, 1)]
+void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
+{
+    float s = 0;
+
+    float1x1 a1x1 = 1 << 0;
+    float1x1 b1x1 = 0;
+    float1x1 c1x1 = select(bool1x1(true), a1x1, b1x1);
+    s += c1x1[0][0];
+
+    float1x2 a1x2 = 1 << 1;
+    float1x2 b1x2 = 0;
+    float1x2 c1x2 = select(bool1x2(true), a1x2, b1x2);
```

Comment:
The test case should have `condition` being a mixed matrix of `true` and `false` instead of just being `true`.